### PR TITLE
Prevent "Options" propsheet from reverting cursor shape settings

### LINKED ISSUE #1219: Console Terminal "Cursor Shape" gets reset when opening "Options" propsheet
Moved from MSFT:21001042

1) go to properties / terminal

2) select vertical bar

3) click ok

 

at this point everything is good.

 

1) go to properties / terminal

2) go to option tab

3) go to terminal tab

 

at this point you'll see that the cursor shape is reset to "legacy style". this only happens if you go to the "options tab". i think this happens because the options page as the "cursor size" option that is interfering -- just a guess.

### PR DESCRIPTION
**Pull request by zadjii-msft**
## Summary of the Pull Request

Whenever you'd open the "options" propsheet, we'd always revert the "cursor shape" to "legacy". This is obviously bad. This PR will make is so the cursor shape will persist, and only selecting a size manually will set the shape to "legacy".

## PR Checklist
* [x] Closes #1219
* [x] I work here
* [n/a] Tests added/passed
* [n/a] Requires documentation to be updated

## Detailed Description of the Pull Request / Additional comments
We actually need to add a hidden radio button to the "size" radio buttons on the options page, which acts as a placeholder for "non-legacy". Whenever we set the shape to not-legacy, we'll check that hidden radio button. Otherwise, whenever the Options page is activated, the selected radio button from each group gets a `BN_CLICKED` notification, and if there's no button selected, the _first_ will still get that notification.

## Validation Steps Performed
I've opened the propsheet and really mucked with it a bunch. This feels right now.

### CODE REVIEW

File: src/propsheet/TerminalPage.cpp [hunk 1]
```diff
@@ -323,10 +323,41 @@ bool TerminalDlgCommand(const HWND hDlg, const WORD item, const WORD command) no
...
+        if (g_hOptionsDlg != INVALID_HANDLE_VALUE)
+        {
+            unsigned int newRadioValue = IDD_CURSOR_ADVANCED;
+            // Check the radio button that would be checked for the current
+            // CursorType/CursorSize
+            if (gpStateInfo->CursorType != 0)
+            {
+                newRadioValue = IDD_CURSOR_ADVANCED;
+            }
+            else if (gpStateInfo->CursorSize <= 25)
```
Comment (DHowett-MSFT) [inline src/propsheet/TerminalPage.cpp]: do we need a helper that turns a cursor height into an IDD_CURSOR_X?

File: src/propsheet/console.rc [hunk 2]
```diff
@@ -42,6 +42,7 @@ BEGIN
                     WS_TABSTOP | WS_GROUP
     AUTORADIOBUTTON "&Medium", IDD_CURSOR_MEDIUM, 14, 33, 84, 10,
     AUTORADIOBUTTON "&Large", IDD_CURSOR_LARGE, 14, 43, 84, 10,
+    AUTORADIOBUTTON "", IDD_CURSOR_ADVANCED, 14, 53, 0, 0,
```
Comment (DHowett-MSFT) [inline src/propsheet/console.rc]: worth a comment here,if this file format supports comments?

**Comment by DHowett-MSFT** [comment]
Oh, one last thing. Would you check whether the new hidden radio button is in the tab stop order? It probably shouldn’t be if it is!

**Comment by zadjii-msft** [comment]
@DHowett-MSFT Already checked, it isn't :)

**Pull request merged**